sdk/loadtestservice/load-testing-rest/src/beginFileValidation.ts

```diff
@@ -0,0 +1,178 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AbortController, AbortSignalLike } from "@azure/abort-controller";
+import { CancelOnProgress, OperationState, SimplePollerLike } from "@azure/core-lro";
+import { FileUploadAndValidatePoller, PolledOperationOptions } from "./models";
+import { AzureLoadTestingClient } from "./index.js";
+import { TestUploadFile201Response } from "./responses";
+import { isUnexpected } from "./isUnexpected";
+import { ReadStream } from "fs";
+
+/**
+ * Submits a Notification Hub job and creates a poller to poll for results.
+ * @param client - The Load Testing client.
+ * @param options - The operation options.
+ * @returns A poller which can be called to poll until completion of the job.
+ */
+export async function beginFileValidation(
+  client : AzureLoadTestingClient,
+  testId: string,
+  fileName: string,
+  file: ReadStream,
+  polledOperationOptions: PolledOperationOptions = {}
+): Promise<FileUploadAndValidatePoller> {
+  const fileUploadResult = await client
+    .path("/tests/{testId}/files/{fileName}", testId, fileName)
+    .put({
+      contentType: "application/octet-stream",
+      body: file,
+    });
+
+  if (isUnexpected(fileUploadResult)) {
+    throw fileUploadResult.body.error;
+  }
+  let fileValidationResponse : TestUploadFile201Response;
+
+  type Handler = (state: OperationState<TestUploadFile201Response>) => void;
+
+  const state: OperationState<TestUploadFile201Response> = {
+    status: "notStarted",
+  };
+
+  const progressCallbacks = new Map<symbol, Handler>();
+  const processProgressCallbacks = async (): Promise<void> =>
+    progressCallbacks.forEach((h) => h(state));
+  let resultPromise: Promise<TestUploadFile201Response> | undefined;
+  let cancelJob: (() => void) | undefined;
+  const abortController = new AbortController();
+  const currentPollIntervalInMs = polledOperationOptions.updateIntervalInMs ?? 2000;
+
+  const poller: SimplePollerLike<OperationState<TestUploadFile201Response>, TestUploadFile201Response> = {
+    async poll(_options?: { abortSignal?: AbortSignalLike }): Promise<void> {
+
+      await client.path("/tests/{testId}/files/{fileName}", testId, fileName).get();
+      if (fileValidationResponse.body.validationStatus === "VALIDATION_INITIATED" || fileValidationResponse.body.validationStatus === "NOT_VALIDATED") {
+        state.status = "running";
+      }
+
+      if (fileValidationResponse.body.validationStatus === "VALIDATION_SUCCESS" || fileValidationResponse.body.validationStatus === "VALIDATION_NOT_REQUIRED") {
+        state.status = "succeeded";
+        state.result = fileValidationResponse;
+      }
+
+      if (fileValidationResponse.body.validationStatus === "VALIDATION_FAILURE") {
+        state.status = "failed";
+        state.error = new Error(fileValidationResponse.body.validationStatus);
+      }
+
+      await processProgressCallbacks();
+
+      if (state.status === "canceled") {
```

Comment:
is there a case where validation is canceled? is so, could you handle it properly and set the status field appropriately?